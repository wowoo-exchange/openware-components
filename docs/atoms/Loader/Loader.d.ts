import * as React from 'react';
interface LoaderProps {
    /**
     * Additional classname
     */
    className?: string;
}
declare const Loader: React.SFC<LoaderProps>;
export { Loader, LoaderProps, };
