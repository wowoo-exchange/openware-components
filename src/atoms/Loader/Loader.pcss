.cr-loader {
  position: relative;
  top: 50%;
  left: 50%;
  width: 3.4em;
  height: 3.4em;
  margin-left: -1.7em;
  margin-top: -0.1em;

  .loader-block {
    position: absolute;
    top: 0;
    left: 0;
    display: inline-block;
    opacity: 0;
    width: 1em;
    height: 1em;
    background: #28334e;
    animation: show 0.9s step-end infinite alternate, pulse 0.9s linear infinite alternate;
  }

  .loader-block:nth-child(1) {
    transform: translate(0, 0);
    animation-delay: 0.1s;
  }

  .loader-block:nth-child(2) {
    transform: translate(1.2em, 0);
    animation-delay: 0.2s;
  }

  .loader-block:nth-child(3) {
    transform: translate(2.4em, 0);
    animation-delay: 0.3s;
  }

  .loader-block:nth-child(4) {
    transform: translate(0, 1.2em);
    animation-delay: 0.5s;
  }

  .loader-block:nth-child(5) {
    transform: translate(1.2em, 1.2em);
    animation-delay: 0.2s;
  }

  .loader-block:nth-child(6) {
    transform: translate(2.4em, 1.2em);
    animation-delay: 0.7s;
  }

  .loader-block:nth-child(7) {
    transform: translate(0, 2.4em);
    animation-delay: 0.6s;
  }

  .loader-block:nth-child(8) {
    transform: translate(1.2em, 2.4em);
    animation-delay: 0.4s;
  }

  .loader-block:nth-child(9) {
    transform: translate(2.4em, 2.4em);
  }
}

@keyframes pulse {
  from,
  40% {
    background: #28334e;
  }

  to {
    background: #2e4c80;
  }
}

@keyframes show {
  from,
  40% {
    opacity: 0;
  }

  41%,
  to {
    opacity: 1;
  }
}
