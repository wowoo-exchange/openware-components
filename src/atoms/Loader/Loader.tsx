import cr from 'classnames';
import * as React from 'react';

interface LoaderProps {
    /**
     * Additional classname
     */
    className?: string;
}

// tslint:disable-next-line
const Loader: React.SFC<LoaderProps> = ({className}) => {
    const classNames = cr('cr-loader', className);
    return (
        <div className={classNames}>
            <span className="loader-block"/>
            <span className="loader-block"/>
            <span className="loader-block"/>
            <span className="loader-block"/>
            <span className="loader-block"/>
            <span className="loader-block"/>
            <span className="loader-block"/>
            <span className="loader-block"/>
            <span className="loader-block"/>
        </div>
    );
};

export {
    Loader,
    LoaderProps,
};
