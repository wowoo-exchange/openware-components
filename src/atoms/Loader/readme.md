Example of usage:

```js
<Loader />
```
