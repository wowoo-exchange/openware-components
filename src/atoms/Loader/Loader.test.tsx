import { shallow } from 'enzyme';
import * as React from 'react';
import { Loader } from './Loader';

describe('Loader', () => {
    it('should render', () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper.hasClass('cr-loader')).toBeTruthy();
    });


    it('should have correct className and size with another props', () => {
        const wrapper = shallow(<Loader {...{className: 'test'}}/>);
        expect(wrapper.hasClass('cr-loader test')).toBeTruthy();
    });
});
