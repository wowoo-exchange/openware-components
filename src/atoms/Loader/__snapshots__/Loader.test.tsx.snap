// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render 1`] = `
<div
  className="cr-loader"
>
  <span
    className="loader-block"
  />
  <span
    className="loader-block"
  />
  <span
    className="loader-block"
  />
  <span
    className="loader-block"
  />
  <span
    className="loader-block"
  />
  <span
    className="loader-block"
  />
  <span
    className="loader-block"
  />
  <span
    className="loader-block"
  />
  <span
    className="loader-block"
  />
</div>
`;
